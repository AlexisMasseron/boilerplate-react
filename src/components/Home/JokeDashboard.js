// Libs
import React from "react";
import PropTypes from "prop-types";
// Material Components
import { Paper, Button, Typography } from "@material-ui/core";
// Styles
import { withStyles } from "@material-ui/core/styles";

const styles = theme => ({
    root: {
        margin: "50px auto",
        display: "flex",
        flexDirection: "column",
        alignItems: "center"
    },
    jokeBox: {
        width: "40%",
        padding: "20px",
        margin: "20px",
        height: "50px",
        display: "flex",
        alignItems: "center",
        justifyContent: "center"
    }
});

const JokeDashboard = ({ classes, currentJoke, fetchJoke }) => (
    <div className={classes.root}>
        <Button variant="raised" color="primary" onClick={fetchJoke}>
            Give me a new joke
        </Button>
        {currentJoke && (
            <Paper className={classes.jokeBox}>
                <Typography>{currentJoke}</Typography>
            </Paper>
        )}
    </div>
);

JokeDashboard.propTypes = {
    classes: PropTypes.object.isRequired,
    currentJoke: PropTypes.string.isRequired,
    fetchJoke: PropTypes.func.isRequired
};

export default withStyles(styles)(JokeDashboard);
