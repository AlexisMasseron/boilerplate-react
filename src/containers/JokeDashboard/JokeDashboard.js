// Libs
import { connect } from "react-redux";

// Actions
import { fetchJoke } from "../../redux/actions/jokes";

// Component
import JokeDashboard from "../../components/Home/JokeDashboard";

const mapStateToProps = state => ({
    currentJoke: state.data.jokes.currentJoke
});

const mapDispatchToProps = {
    fetchJoke
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(JokeDashboard);
