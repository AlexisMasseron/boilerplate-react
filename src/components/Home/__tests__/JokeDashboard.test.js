import React from "react";
import toJson from "enzyme-to-json";

import JokeDashboard from "../JokeDashboard";
import createShallow from "../../../tests-utils/createShallow";

describe("<Home />", () => {
    let shallow;
    let wrapper;

    beforeAll(() => {
        shallow = createShallow({ dive: 1 });
    });
    beforeEach(() => {
        wrapper = shallow(<JokeDashboard />);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
