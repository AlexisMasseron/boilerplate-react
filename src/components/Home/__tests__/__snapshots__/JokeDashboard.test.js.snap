// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
<div
  className="JokeDashboard-root-1"
>
  <WithStyles(Button)
    color="primary"
    onClick={[Function]}
    variant="raised"
  >
    Give me a new joke
  </WithStyles(Button)>
</div>
`;
